### PR TITLE
More sonarqube fixes

### PR DESCRIPTION
**Pull request by tsatam**
## Overview
Fixes more SonarQube violations, primarily those regarding the use of JPA entities as request bodies.

### CODE REVIEW

File: api/src/main/java/com/ford/labs/retroquest/actionitem/CreateActionItemRequest.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+import java.sql.Date;
+
+@Value
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class CreateActionItemRequest {
+    String task;
+    boolean completed;
+    String assignee;
+    @EqualsAndHashCode.Exclude
```
Comment (LowBudgetMan) [inline api/src/main/java/com/ford/labs/retroquest/actionitem/CreateActionItemRequest.java:15]: Not something we need to fix with this PR, but would be curious to have a conversation around why we'd want to exclude this from equals and hashcode.

File: api/src/test/java/com/ford/labs/retroquest/deprecated_tests/BoardServiceTest.java [hunk 2]
```diff
@@ -103,21 +108,50 @@ void getBoardsForTeamId_shouldReturnAPagedResult() {
...
+                    boardId
+                );
+            });
+
+        var returnedBoard = boardService.createBoard(boardToSave);
+
+        assertThat(returnedBoard.getId()).isEqualTo(1234L);
+        assertThat(returnedBoard.getTeamId()).isEqualTo(expectedTeamId);
+        assertThat(returnedBoard.getThoughts()).hasSize(1);
+        assertThat(returnedBoard.getThoughts()).hasSize(1);
```
Comment (LowBudgetMan) [inline api/src/test/java/com/ford/labs/retroquest/deprecated_tests/BoardServiceTest.java:154]: Why do we check the size of the thoughts twice?

**Review by LowBudgetMan — CHANGES_REQUESTED** [review]
We should remove that duplicate assertion while we're here since it's not changing functionality. Other than that it looks good.

File: api/src/test/java/com/ford/labs/retroquest/deprecated_tests/BoardServiceTest.java (hunk 2 above)
Comment (tsatam) [reply api/src/test/java/com/ford/labs/retroquest/deprecated_tests/BoardServiceTest.java]: Typo. I've fixed it.

**Review by LowBudgetMan — APPROVED** [review]
:shipit:

**Pull request merged**